File: source/53.最大子序和.cpp
/*
 * @lc app=leetcode.cn id=53 lang=cpp
 *
 * [53] 最大子序和
 */
#include "cpp_includes.h"
// @lc code=start
class Solution {
 public:
  int maxSubArray(vector<int>& nums) {
    int f = nums[0];
    int res = nums[0];

    for (int i = 1; i < nums.size(); i++) {
      f = max(f + nums[i], nums[i]);
      res = max(res, f);
    }
    return res;
  }
};
// @lc code=end

File: 53.最大子序和.py
#
# @lc app=leetcode.cn id=53 lang=python3
#
# [53] 最大子序和
#

# @lc code=start
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        Sum = nums[0]
        Max = nums[0]
        for i in range(1, len(nums)):
            if Sum + nums[i] < nums[i] :
                Sum = nums[i]
            else:
                Sum += nums[i]

            Max = max(Max, Sum)

        return Max 

# @lc code=end

